lib/gateway.js

```diff
@@ -90,58 +90,82 @@ function defineResolvers (schema, typeToServiceMap, serviceMap, typeFieldsToServ
           const serviceForFieldType = typeToServiceMap[fieldType]
-
-          if (serviceForFieldType !== serviceForType && !valueTypes.includes(`${fieldType.name}`)) {
-            /* istanbul ignore else */
-            if (serviceForType === null) {
-              // Either query, mutation, subscription
-              if (type.name === 'Subscription') {
-                field.subscribe = makeResolver({
-                  service: serviceMap[serviceForFieldType],
-                  createOperation: createQueryOperation,
-                  isQuery: true,
-                  isSubscription: true
-                })
-              } else {
+          /* istanbul ignore else */
+          if (
+            (serviceForFieldType === null && serviceForType !== null) ||
+            (serviceForFieldType !== null && serviceForType !== null && serviceForFieldType === serviceForType)
+          ) {
+            /**
+             * Either there is a service for the type and no service for the field type,
+             * hence the field type is a value type and its not a query / mutation / subscription.
+             * Or there is a service for the type and the field type and it is the same service.
+             * In both cases, we resolve from the parent
+             */
+            field.resolve = (parent, args, context, info) => parent && parent[info.path.key]
+          } else if (serviceForType === null) {
+            /**
+             * If there is no service for the type, it is a query, mutation or subscription
+             */
+            let service = serviceMap[serviceForFieldType]
+            if (typeof service === 'undefined') {
+              /**
+               * If the value type is the return type of a query, subscription or mutation, its service is undefined, e.g. for
+               * extend type Query {
+               *    userServiceInfo: SomeReturnType!
+               * }
+               * where SomeReturnType is a value type.
+               * In these cases, we get the service from the typeFieldsToService map.
+               */
+              service = serviceMap[typeFieldsToService[`${type}-${fieldName}`]]
+            }
+            if (type.name === 'Subscription') {
+              field.subscribe = makeResolver({
+                service: service,
+                createOperation: createQueryOperation,
+                isQuery: true,
+                isSubscription: true
+              })
+            } else {
+              field.resolve = makeResolver({
+                service: service,
+                createOperation: createQueryOperation,
+                transformData: response => response.json.data[fieldName],
+                isQuery: true
+              })
+            }
+          } else if (serviceForFieldType !== null && serviceForFieldType !== serviceForType) {
+            /**
+             * If there is no service for the field type but a service for the type and it is not the same,
```

Comment:
this should be `If there is a service for the field type`

Reply:
can you update it?

Reply:
Yes, I also saw that the docs on function level are not required anymore. Should I remove them or rather change it to reflect the new design?

Reply:
Updated them to the new design

Reply:
I updated the docs :)